app/code/core/Mage/Adminhtml/controllers/Promo/CatalogController.php

```diff
@@ -308,6 +308,38 @@ public function applyRulesAction()
         $this->_redirect('*/*');
     }
 
+     /**
+     * Duplicate catalog rule and edit
+     */
+    public function duplicateAction()
+    {
+        $errorMessage = Mage::helper('catalogrule')->__('Cannot copy rule ');
+        $params = $this->getRequest()->getParams();
+        try {
+            if($model = Mage::getModel('catalogrule/rule')->load($params['id'])) {
```

Comment:
The return value of `load()` is an object so always truthy. The assignment should go before the `if` clause and then `getId` should be used to see if the load was successful.